### PR TITLE
feat(core): Add FsModuleLoader that supports loading from filesystem

### PR DESCRIPTION
**Pull request by magurotuna**
This commit adds `FsModuleLoader` to `deno_core`, which implements `ModuleLoader` trait.
It is used when creating a runtime that supports module loading from filesystem. In particular, it's going to be used by `deno_lint` plugin feature. 

Ref: 
- https://github.com/denoland/deno_lint/pull/415
- https://github.com/denoland/deno/pull/8381

CC @bartlomieju

### CODE REVIEW

File: core/modules.rs [hunk 1]
```diff
@@ -130,6 +130,43 @@ impl ModuleLoader for NoopModuleLoader {
...
+    _op_state: Rc<RefCell<OpState>>,
+    module_specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<ModuleSpecifier>,
+    _is_dynamic: bool,
+  ) -> Pin<Box<ModuleSourceFuture>> {
+    let module_specifier = module_specifier.clone();
+    async move {
+      let path = module_specifier.as_url().to_file_path().unwrap();
+      let content = std::fs::read_to_string(path)?;
+      let module = ModuleSource {
```
Comment (rheh) [inline core/modules.rs:159]: Could these three let be const?

File: core/modules.rs [hunk 2]
```diff
@@ -130,6 +130,43 @@ impl ModuleLoader for NoopModuleLoader {
...
+
+  fn load(
+    &self,
+    _op_state: Rc<RefCell<OpState>>,
+    module_specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<ModuleSpecifier>,
+    _is_dynamic: bool,
+  ) -> Pin<Box<ModuleSourceFuture>> {
+    let module_specifier = module_specifier.clone();
+    async move {
```
Comment (rheh) [inline core/modules.rs:156]: why async?  I see no await within?

File: core/modules.rs [hunk 3]
```diff
@@ -130,6 +130,43 @@ impl ModuleLoader for NoopModuleLoader {
...
+      let path = module_specifier.as_url().to_file_path().unwrap();
+      let content = std::fs::read_to_string(path)?;
```
Comment (rheh) [inline core/modules.rs:158]: Renamed to content -> code?

File: core/modules.rs [hunk 4]
```diff
@@ -130,6 +130,43 @@ impl ModuleLoader for NoopModuleLoader {
...
+      let module = ModuleSource {
+        code: content,
```
Comment (rheh) [inline core/modules.rs:160]: If we renamed to code, we could just:
`{
code,
...
}`

File: core/modules.rs (hunk 2 above)
Comment (bartlomieju) [reply core/modules.rs:159]: The trait requires to return future from this method.

File: core/modules.rs (hunk 1 above)
Comment (bartlomieju) [reply core/modules.rs:167]: No, Rust doesn't allow that

**Comment by bartlomieju** [comment]
@magurotuna I applied some changes to error handling and doc comment. I'll wait on @ry's review before landing

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**